scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -0,0 +1,1226 @@
+        return WrappedMultiplexedStreamResponse( self, response )
+
+    def _get_raw_response_socket(self, response):
+        if response.raw._fp.fp:
+            return super( DockerClient, self )._get_raw_response_socket( response )
+
+        return None
+
+def _split_datetime_from_line( line ):
+    """Docker timestamps are in RFC3339 format: 2015-08-03T09:12:43.143757463Z, with everything up to the first space
+    being the timestamp.
+    """
+    log_line = line
+    dt = datetime.datetime.utcnow()
+    pos = line.find( ' ' )
+    if pos > 0:
+        dt = scalyr_util.rfc3339_to_datetime( line[0:pos] )
+        log_line = line[pos+1:]
+
+    return (dt, log_line)
+
+def _get_containers(client, ignore_container=None, restrict_to_container=None, logger=None,
+                    only_running_containers=True, glob_list=None, include_log_path=False):
+    """Gets a dict of running containers that maps container id to container name
+    """
+    if logger is None:
+        logger = global_log
+
+    result = {}
+    try:
+        filters = {"id": restrict_to_container} if restrict_to_container is not None else None
+        response = client.containers(filters=filters, all=not only_running_containers)
+        for container in response:
+            cid = container['Id']
+
+            if ignore_container is not None and cid == ignore_container:
+                continue
+
+            if len( container['Names'] ) > 0:
+                name = container['Names'][0].lstrip('/')
+
+                add_container = True
+
+                if glob_list:
+                    add_container = False
+                    for glob in glob_list:
+                        if fnmatch.fnmatch( name, glob ):
+                            add_container = True
+                            break;
+
+                if add_container:
+                    log_path = None
+                    if include_log_path:
+                        try:
+                            info = client.inspect_container( cid )
+                            log_path = info['LogPath'] if 'LogPath' in info else None
+                        except Exception, e:
+                          logger.error("Error inspecting container '%s'" % cid, limit_once_per_x_secs=300,limit_key="docker-api-inspect")
+
+                    result[cid] = {'name': name, 'log_path': log_path }
+
+            else:
+                result[cid] = {'name': cid, 'log_path': None}
+
+    except Exception, e:  # container querying failed
+        logger.error("Error querying running containers", limit_once_per_x_secs=300,
+                     limit_key='docker-api-running-containers' )
+        result = None
+
+    return result
+
+class PodInfo( object ):
+    """
+        A collection class that stores label and other information about a kubernetes pod
+    """
+    def __init__( self, name='', namespace='', uid='', nodeName='', labels={}, containers=[] ):
```

Comment:
Can we change to `node_name`?  We should be consistent about variable naming convention.